pkg/ingestor/verifier/verifier_test.go

```diff
@@ -0,0 +1,164 @@
+			"invocation": {
+			  "configSource": {
+				"uri": "git+https://github.com/curl/curl-docker@master",
+				"digest": { "sha1": "d6525c840a62b398424a78d792f457477135d0cf" },   
+				"entryPoint": "build.yaml:maketgz"
+			  }
+			},
+			"metadata": {
+			  "buildStartedOn": "2020-08-19T08:38:00Z",
+			  "completeness": {
+				  "environment": true
+			  }
+			},
+			"materials": [
+			  {
+				"uri": "git+https://github.com/curl/curl-docker@master",
+				"digest": { "sha1": "d6525c840a62b398424a78d792f457477135d0cf" }
+			  }, {
+				"uri": "github_hosted_vm:ubuntu-18.04:20210123.1",
+				"digest": { "sha1": "d6525c840a62b398424a78d792f457477135d0cf" }
+			  }
+			]
+		}
+	}`
+	b64ITE6SLSA    = base64.StdEncoding.EncodeToString([]byte(ite6SLSA))
+	ite6Payload, _ = json.Marshal(dsse.Envelope{
+		PayloadType: "https://in-toto.io/Statement/v0.1",
+		Payload:     b64ITE6SLSA,
+		Signatures: []dsse.Signature{{
+			KeyID: "id1",
+			Sig:   "test",
+		}},
+	})
+	ite6DSSEDoc = processor.Document{
+		Blob:   ite6Payload,
+		Type:   processor.DocumentDSSE,
+		Format: processor.FormatJSON,
+		SourceInformation: processor.SourceInformation{
+			Collector: "TestCollector",
+			Source:    "TestSource",
+		},
+	}
+	unknownDoc = processor.Document{
+		Blob:   []byte("fake payload"),
+		Type:   processor.DocumentUnknown,
+		Format: processor.FormatJSON,
+		SourceInformation: processor.SourceInformation{
+			Collector: "TestCollector",
+			Source:    "TestSource",
+		},
+	}
+	ecdsaPubKey, pemBytes, _ = keyutil.GetECDSAPubKey()
+)
+
+type mockSigstoreVerifier struct{}
+
+func newMockSigstoreVerifier() *mockSigstoreVerifier {
+	return &mockSigstoreVerifier{}
+}
+
+func (m *mockSigstoreVerifier) Verify(payloadBytes []byte) ([]Identity, error) {
+	h := sha256.Sum256(pemBytes)
+	return []Identity{
+		{
+			ID: "test",
+			Key: key.Key{
+				Hash:   "SHA256" + ":" + hex.EncodeToString(h[:]),
+				Type:   "ecdsa",
+				Val:    ecdsaPubKey,
+				Scheme: "ecdsa",
+			},
+			Verified: true,
+		},
+	}, nil
+}
+
+func (m *mockSigstoreVerifier) Type() VerifierType {
+	return "sigstore"
+}
+
+func TestVerifyIdentity(t *testing.T) {
+	RegisterVerifier(newMockSigstoreVerifier(), "sigstore")
+
+	h := sha256.Sum256(pemBytes)
+	tests := []struct {
+		name    string
+		doc     *processor.Document
+		want    []Identity
+		wantErr bool
+	}{{
+		name: "verify",
+		doc:  &ite6DSSEDoc,
+		want: []Identity{
+			{
+				ID: "test",
+				Key: key.Key{
+					Hash:   "SHA256" + ":" + hex.EncodeToString(h[:]),
+					Type:   "ecdsa",
+					Val:    ecdsaPubKey,
+					Scheme: "ecdsa",
+				},
+				Verified: true,
+			},
+		},
+		wantErr: false,
+	}, {
+		name:    "not found",
+		doc:     &unknownDoc,
+		wantErr: true,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := VerifyIdentity(tt.doc)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("VerifyIdentity() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
if its an error case, it should not check this

Reply:
done.